implementations/rust/ockam/ockam_node/src/storage/database/migrations/migration_support/migrator.rs

```diff
@@ -110,94 +121,8 @@ impl Migrator {
-                }
-            }
-        }
-
-        let sql_iterator = self.sql_migrator.migrations.iter().filter_map(|m| {
-            let version = m.version;
-
-            if !Self::is_inside_version_range(version, from_version, to_version, run_last_sql) {
-                return None;
-            }
-
-            Some((version, NextMigration::Sql(m)))
-        });
-        let rust_iterator = self.rust_migrations.iter().filter_map(|m| {
-            let version = m.version();
-
-            if !Self::is_inside_version_range(version, from_version, to_version, false) {
-                return None;
-            }
-
-            Some((version, NextMigration::Rust(m)))
-        });
-
-        let mut all_migrations: Vec<(i64, NextMigration)> =
-            sql_iterator.chain(rust_iterator).collect();
-        all_migrations.sort_by(|m1, m2| match m1.0.cmp(&m2.0) {
-            Ordering::Less => Ordering::Less,
-            Ordering::Equal => {
-                // Sql migrations go first
-                if m1.1.is_sql() {
-                    Ordering::Less
-                } else {
-                    Ordering::Greater
-                }
-            }
-            Ordering::Greater => Ordering::Greater,
-        });
-
-        for migration in all_migrations.iter().map(|(_version, m)| m) {
-            match migration {
-                NextMigration::Sql(sql_migration) => {
-                    if sql_migration.migration_type.is_down_migration() {
-                        return Ok(());
-                    }
-
-                    match applied_migrations.get(&sql_migration.version) {
-                        Some(applied_migration) => {
-                            if sql_migration.checksum != applied_migration.checksum {
-                                return Err(ockam_core::Error::new(
-                                    Origin::Node,
-                                    Kind::Conflict,
-                                    format!(
-                                        "Checksum mismatch for sql migration for version {}",
-                                        sql_migration.version
-                                    ),
-                                ));
-                            }
-                        }
-                        None => {
-                            connection.apply(sql_migration).await.into_core()?;
-                        }
-                    }
-                }
-                NextMigration::Rust(rust_migration) => {
-                    if Self::has_migrated(connection, rust_migration.name()).await? {
-                        continue;
-                    }
-                    if rust_migration.migrate(connection).await? {
-                        Self::mark_as_migrated(connection, rust_migration.name()).await?;
-                    }
-                }
-            }
+        for migration in migrations.into_iter() {
+            migration.apply(connection, &applied_migrations).await?;
```

Comment:
I would inline the implementation here. Otherwise, it's unclear that `applied_migrations` only include sql migrations, while rust migrations also have their own mechanism of checking if they were applied.